src/overlays/actors/ovl_Boss_Ganondrof/z_boss_ganondrof.c

```diff
@@ -61,71 +143,1400 @@ static ColliderCylinderInit D_80914CFC = {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+    }
+    if ((this->actionFunc == BossGanondrof_Stunned) && (this->timers[0] > 1)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+    } else if (this->actionFunc == BossGanondrof_Block) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+    } else if (this->actionFunc == BossGanondrof_Charge) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderSpear.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/func_809106F0.s")
+    this->actor.focus.pos = this->targetPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/func_80910738.s")
+    sp70 = Math_SinS(-this->actor.shape.rot.y);
+    sp74 = Math_CosS(-this->actor.shape.rot.y);
+    sp6C = ((sp70 * this->actor.velocity.z) + (sp74 * this->actor.velocity.x)) * 300.0f;
+    sp68 = ((-sp70 * this->actor.velocity.x) + (sp74 * this->actor.velocity.z)) * 300.0f;
+    Math_ApproachF(&this->legRotY, sp6C, 1.0f, 600.0f);
+    Math_ApproachF(&this->legRotZ, sp68, 1.0f, 600.0f);
+    if ((this->flyMode != FHG_FLY_PAINTING) && (this->actionFunc != BossGanondrof_Stunned) &&
+        (this->deathState == NOT_DEAD)) {
+        sp64 = (Math_SinS(this->varianceTimer * 0x8DC) * -500.0f) - 500.0f;
+    } else {
+        sp64 = 0.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/func_80910784.s")
+    Math_ApproachF(&this->legSplitY, sp64, 1.0f, 100.0f);
+    if (this->shockTimer != 0) {
+        this->shockTimer--;
+        osSyncPrintf("F 1\n");
+        for (j = 0; j < 7; j++) {
+            osSyncPrintf("F 15\n");
+            EffectSsFhgFlash_SpawnShock(globalCtx, &this->actor, &this->actor.world.pos, 45, FHGFLASH_SHOCK_PG);
+        }
+        osSyncPrintf("F 2\n");
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/func_80910A34.s")
+    if (this->actor.params == FHG_REAL_BOSS) {
+        Lights_PointNoGlowSetInfo(&this->lightInfo, this->spearTip.x, this->spearTip.y, this->spearTip.z, 255, 255, 255,
+                                  200);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/BossGanondrof_Init.s")
+s32 BossGanondrof_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                   void* thisx) {
+    BossGanondrof* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganondrof/BossGanondrof_Destroy.s")
+    switch (limbIndex) {
+        case 14 + 1:
```

Comment:
What's the reason to write all these with + 1 here? Eventually these limb indexes will have an associated enum like PlayerLimb, and all their values will already include the + 1

Reply:
It made more sense before we had all this knowledge of how skeletons work.